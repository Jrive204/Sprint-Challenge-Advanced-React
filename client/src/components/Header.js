import React, { useState } from "react";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
import { useStyles } from "./styles/HeaderStyles";
import { useDarkMode } from "./hooks/useDarkMode";
import { PurpleSwitch } from "./styles/HeaderStyles";

export default function Header() {
  const [toggleMode] = useDarkMode("dark", false);
  const [state, setState] = useState({
    checkedA: true
  });

  const handleChange = name => event => {
    setState({ ...state, [name]: event.target.checked });
  };

  const classes = useStyles();

  return (
    <div className={classes.root}>
      <AppBar position='static'>
        <Toolbar>
          <IconButton
            edge='start'
            className={classes.menuButton}
            color='inherit'
            aria-label='menu'>
            <MenuIcon />
          </IconButton>
          <Typography variant='h6' className={classes.title}></Typography>

          <PurpleSwitch
            value='checkedA'
            onClick={handleChange("checkedA")}
            checked={state.checkedA}
            onChange={toggleMode}
          />
        </Toolbar>
      </AppBar>
    </div>
  );
}
