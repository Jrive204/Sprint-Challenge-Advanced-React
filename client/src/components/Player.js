import React from "react";

export default function Player(props) {
  return (
    <div>
      <p>Player {console.log(props, "player")}</p>
    </div>
  );
}
