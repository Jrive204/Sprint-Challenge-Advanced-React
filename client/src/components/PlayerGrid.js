import React from "react";
import noImage from "../img/no_image2.png";
// import Player from './Player'

export default function PlayerGrid(props) {
  return (
    <div data-testid='PlayerGrid' className='PlayerGrid'>
      {props.player
        .map(players => (
          <div key={players.id} className='flip-card'>
            <div className='flip-card-inner'>
              <div className='flip-card-front'>
                <img
                  src={noImage}
                  alt='Avatar'
                  style={{ width: "200px", height: "200px" }}
                />
              </div>
              <div className='flip-card-back'>
                <h1>{players.name}</h1>
                <p>{players.country}</p>
                <p>Searches: {players.searches}</p>
              </div>
            </div>
          </div>
        ))
        .sort()}
    </div>
  );
}
