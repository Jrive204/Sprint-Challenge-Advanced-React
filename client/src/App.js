import React, { Component } from "react";
import axios from "axios";
import Header from "./components/Header";
import PlayerGrid from "./components/PlayerGrid";
import Plot from "react-plotly.js";

export default class App extends Component {
  state = {
    player: []
  };
  componentDidMount() {
    axios
      .get("http://localhost:5000/api/players")
      .then(res => {
        console.log(res.data, `data`);
        this.setState({
          player: res.data
        });
      })
      .catch(err => console.log(err));
  }

  render() {
    let playerid = this.state.player.map(p => {
      return p.id;
    });
    let playersearch = this.state.player.map(p => p.searches);
    let playername = this.state.player.map(p => p.name);
    console.log(playerid, "playerid");
    return (
      <div className='App'>
        <Header />
        <h1 style={{ textAlign: "center" }}>
          Women's World Cup players ranked by search interest from Google Trends
        </h1>
        <PlayerGrid player={this.state.player} />
        <div className='plotgraph'>
          <Plot
            data={[
              {
                type: "bar",
                x: playerid,
                y: playersearch,
                marker: { color: "purple" },
                name: playername
              }
            ]}
            layout={{
              title: "Woman World Cup Name Searches",
              font: {
                size: "15"
              },
              xaxis: {
                title: "Players"
              },
              yaxis: {
                title: "Searches"
              },
              autosize: true,
              width: "1500"
            }}
            // style={{ fontSize: "100px" }}
          />
        </div>
      </div>
    );
  }
}
