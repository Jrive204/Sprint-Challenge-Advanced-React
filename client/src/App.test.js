import React from "react";
import { shallow, mount, render } from "enzyme";
import renderer from "react-test-renderer";

import App from "./App";
import PlayerGrid from "./components/PlayerGrid";

import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Player from "./components/Player";
import Header from "./components/Header";

window.DOMParser = function() {};

window.URL.createObjectURL = function() {};

configure({ adapter: new Adapter() });

const player = [
  {
    name: "Alex Morgan",
    country: "United States",
    searches: 100,
    id: 0
  },
  {
    name: "Megan Rapinoe",
    country: "United States",
    searches: 99,
    id: 1
  }
];

describe("App component", () => {
  it("Check if Title is correct", () => {
    const wrapper = shallow(<App />);
    const text = wrapper.find("h1").text();
    expect(text).toEqual(
      "Women's World Cup players ranked by search interest from Google Trends"
    );
  });

  it("Check if header is correct", () => {
    const wrapper = shallow(<Header />);
    const text = wrapper.find("div").text();
    expect(text).toEqual("Women's World Cup Info");
  });
  test("renders plotly", () => {
    const wrapper = shallow(<App />);
    window.location.reload = jest.fn();
    window.location.reload();

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state().data).toEqual(null);
  });

  it("matches the PlayerGrid snapshot", () => {
    const tree = renderer.create(<PlayerGrid player={player} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// describe("Addition", () => {
//   it("knows that 2 and 2 make 4", () => {
//     expect(2 + 2).toBe(4);
//   });
// });
