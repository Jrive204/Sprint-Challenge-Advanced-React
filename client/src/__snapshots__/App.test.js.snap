// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `
<div
  className="App"
>
  <div
    className="makeStyles-root-1"
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="menu"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
        <h6
          className="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
        >
          Women's World Cup Info
        </h6>
        <span
          className="MuiSwitch-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-111 MuiSwitch-switchBase WithStyles(ForwardRef(Switch))-switchBase-96 MuiSwitch-colorSecondary PrivateSwitchBase-checked-112 Mui-checked WithStyles(ForwardRef(Switch))-checked-97"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                checked={true}
                className="PrivateSwitchBase-input-114 MuiSwitch-input"
                disabled={false}
                onChange={[Function]}
                type="checkbox"
                value="checkedA"
              />
              <span
                className="MuiSwitch-thumb"
              />
            </span>
          </span>
          <span
            className="MuiSwitch-track WithStyles(ForwardRef(Switch))-track-98"
          />
        </span>
      </div>
    </header>
  </div>
  <h1
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Women's World Cup players ranked by search interest from Google Trends
  </h1>
  <div
    className="PlayerGrid"
    data-testid="PlayerGrid"
  />
  <div
    className="plotgraph"
  />
</div>
`;

exports[`App component matches the snapshot 2`] = `
<div
  className="PlayerGrid"
  data-testid="PlayerGrid"
>
  <div
    className="flip-card"
  >
    <div
      className="flip-card-inner"
    >
      <div
        className="flip-card-front"
      >
        <img
          alt="Avatar"
          src="no_image2.png"
          style={
            Object {
              "height": "200px",
              "width": "200px",
            }
          }
        />
      </div>
      <div
        className="flip-card-back"
      >
        <h1>
          Alex Morgan
        </h1>
        <p>
          United States
        </p>
        <p>
          Searches: 
          100
        </p>
      </div>
    </div>
  </div>
  <div
    className="flip-card"
  >
    <div
      className="flip-card-inner"
    >
      <div
        className="flip-card-front"
      >
        <img
          alt="Avatar"
          src="no_image2.png"
          style={
            Object {
              "height": "200px",
              "width": "200px",
            }
          }
        />
      </div>
      <div
        className="flip-card-back"
      >
        <h1>
          Megan Rapinoe
        </h1>
        <p>
          United States
        </p>
        <p>
          Searches: 
          99
        </p>
      </div>
    </div>
  </div>
</div>
`;
